storage/akamaistorage.py

```diff
@@ -69,6 +80,26 @@ def get_direct_download_url(
 
         # add akamai signed token
         try:
+            if self.amz_encryption_token is not None and self.amz_encryption_iv is not None:
+                query_params = urllib.parse.parse_qs(akamai_url_parsed.query)
```

Comment:
you want `iv` to be random and added to the message, same thing with the `nonce`

Reply:
there's a field for `iv` on the akamai side too, i assumed they had to match

Reply:
yes, in Akamai, you'll have to extract the nonce and iv from the encrypted message and store them in a variable there. Every request will have a different nonce and an IV

Reply:
so those fields in the akamai dashboard are for the key's to pick up on the iv and nonce? How does akamai find those values?

Reply:
you have to send those values in the message either as another query param

Reply:
how will akamai consume those values though?

Reply:
updated to return unique iv each time